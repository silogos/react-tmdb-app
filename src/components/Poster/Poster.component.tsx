import { PropsWithChildren } from "react";
import { PosterProps } from "./Poster.type";
import { ViewfinderCircleIcon } from "@heroicons/react/24/outline";

function Poster({
  poster,
  className,
  children,
}: PropsWithChildren<PosterProps>) {
  return (
    <div
      className={`relative aspect-[8/12] bg-gray-700 bg-opacity-90 rounded-lg overflow-hidden ${className}`}
    >
      {poster && (
        <img
          className="w-full group-hover:lg:scale-110 duration-150"
          src={`https://image.tmdb.org/t/p/w300/${poster}`}
          alt="Movie Poster"
        ></img>
      )}

      {!poster && (
        <div className="absolute inset-0 flex flex-col justify-center items-center gap-2 text-gray-300 text-center">
          <ViewfinderCircleIcon className="w-1/2" />
          Image Not Found
        </div>
      )}

      {children}
    </div>
  );
}

export default Poster;
