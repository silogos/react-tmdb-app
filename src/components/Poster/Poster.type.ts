export type PosterProps = {
  poster?: string;
  className?: string;
};
