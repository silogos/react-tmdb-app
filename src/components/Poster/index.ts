export { default } from "./Poster.component";
