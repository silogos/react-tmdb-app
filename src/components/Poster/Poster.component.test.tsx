import { describe, it, expect } from "vitest";
import Poster from "./Poster.component";
import { render } from "@testing-library/react";

describe("Poster Component", () => {
  it("renders correctly", () => {
    const component = render(<Poster poster="testPath" />);

    expect(component.container).toMatchSnapshot();
  });

  it("renders correctly without poster image", () => {
    const component = render(<Poster />);

    expect(component.container).toMatchSnapshot();
  });

  it("displays the correct image source", () => {
    const { getByAltText } = render(<Poster poster="testPath" />);
    const imgElement = getByAltText("Movie Poster") as HTMLImageElement;
    expect(imgElement.src).toBe("https://image.tmdb.org/t/p/w300/testPath");
  });
});
