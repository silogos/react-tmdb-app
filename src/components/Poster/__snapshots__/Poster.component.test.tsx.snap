// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Poster Component > renders correctly 1`] = `
<div>
  <div
    class="relative aspect-[8/12] bg-gray-700 bg-opacity-90 rounded-lg overflow-hidden undefined"
  >
    <img
      alt="Movie Poster"
      class="w-full group-hover:lg:scale-110 duration-150"
      src="https://image.tmdb.org/t/p/w300/testPath"
    />
  </div>
</div>
`;

exports[`Poster Component > renders correctly without poster image 1`] = `
<div>
  <div
    class="relative aspect-[8/12] bg-gray-700 bg-opacity-90 rounded-lg overflow-hidden undefined"
  >
    <div
      class="absolute inset-0 flex flex-col justify-center items-center gap-2 text-gray-300 text-center"
    >
      <svg
        aria-hidden="true"
        class="w-1/2"
        data-slot="icon"
        fill="none"
        stroke="currentColor"
        stroke-width="1.5"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M7.5 3.75H6A2.25 2.25 0 0 0 3.75 6v1.5M16.5 3.75H18A2.25 2.25 0 0 1 20.25 6v1.5m0 9V18A2.25 2.25 0 0 1 18 20.25h-1.5m-9 0H6A2.25 2.25 0 0 1 3.75 18v-1.5M15 12a3 3 0 1 1-6 0 3 3 0 0 1 6 0Z"
          stroke-linecap="round"
          stroke-linejoin="round"
        />
      </svg>
      Image Not Found
    </div>
  </div>
</div>
`;
